src/backend/distributed/utils/citus_ruleutils.c

```diff
@@ -229,51 +229,61 @@ pg_get_tableschemadef_string(Oid tableRelationId)
 	AttrNumber constraintIndex = 0;
 	AttrNumber constraintCount = 0;
 	StringInfoData buffer = { NULL, 0, 0, 0 };
+	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
 
-	/*
-	 * Instead of retrieving values from system catalogs as other functions in
-	 * ruleutils.c do, we follow an unusual approach here: we open the relation,
-	 * and fetch the relation's tuple descriptor. We do this because the tuple
-	 * descriptor already contains information harnessed from pg_attrdef,
-	 * pg_attribute, pg_constraint, and pg_class; and therefore using the
-	 * descriptor saves us from a lot of additional work.
-	 */
-	relation = relation_open(tableRelationId, AccessShareLock);
-	relationName = generate_relation_name(tableRelationId, NIL);
+	overridePath->schemas = NIL;
+	overridePath->addCatalog = true;
+	PushOverrideSearchPath(overridePath);
 
-	relationKind = relation->rd_rel->relkind;
-	if (relationKind != RELKIND_RELATION && relationKind != RELKIND_FOREIGN_TABLE)
+	PG_TRY();
```

Comment:
I am not sure if we can avoid that. We need to be sure that PopOverrideSearchPath(); is called even if some error occures during query execution. Otherwise we may corrupt search_path stack. Do you have any suggestion instead of PG_TRY()?